### PR TITLE
Delete EFS

### PR DESCRIPTION
**Pull request by williazz**
# Description of the issue
EFS is not necessary because we are not validating for dimensions `fstype` and `path`

# Description of changes
* Delete EFS from terraform
* Delete EFS scripts from main repo in [separate PR](https://github.com/aws/amazon-cloudwatch-agent/pull/720)

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
[Ran on fork](https://github.com/williazz/amazon-cloudwatch-agent/actions/runs/4442199265)

### CODE REVIEW

**Comment by taohungyang** [comment]
I saw some of the integration tests failed. Did it happen before your change?

**Comment by williazz** [comment]
Those tests like CA_Bundle, ECSEC2IntegrationTest, ECSFargate, etc. never worked on my fork

**Comment by williazz** [comment]
Note: we should run the EFS cleanup once before merging this PR. And also do a manual check

File: terraform/setup/vpc.tf
Comment (williazz) [inline terraform/setup/vpc.tf:43]: This is an unrelated change to fix ECSFargate test on my fork

File: terraform/ec2/README.md
Comment (williazz) [inline terraform/ec2/README.md:201]: Unrelated change to add missing permission to ECSEC2 test

Comment (taohungyang) [reply terraform/ec2/README.md:201]: Also, do we need to remove the elasticfilesystem permissions above? I know this is just a readme and wouldn't affect the existing roles/policies but just want to make sure if those permissions are still required. I'll approve it if that's still in used by some other tests.

Comment (williazz) [reply terraform/ec2/README.md:201]: oh good catch, i forgot to remove those efs permissions

sure I can sort them

**Review by taohungyang — APPROVED** [review]

**Review by khanhntd — APPROVED** [review]

**Pull request merged**
